js/ai/src/model.ts

```diff
@@ -109,6 +109,7 @@ export type Role = z.infer<typeof RoleSchema>;
 export const MessageSchema = z.object({
   role: RoleSchema,
   content: z.array(PartSchema),
+  metadata: z.record(z.unknown()).optional(),
```

Comment:
parts already have metadata: https://github.com/firebase/genkit/blob/83398d6e9762e8c2b21fa4d1de5258229fce499a/js/ai/src/model.ts#L46
Do we want to keep both?

Reply:
I think so, they are different altitudes and so provide opportunities for different types of transformation.